[debug] [2020-12-25T14:44:01.183Z] ----------------------------------------------------------------------
[debug] [2020-12-25T14:44:01.187Z] Command:       C:\Program Files\nodejs\node.exe C:\Users\Eduard\AppData\Roaming\npm\node_modules\firebase-tools\lib\bin\firebase.js deploy --only functions
[debug] [2020-12-25T14:44:01.188Z] CLI Version:   8.12.1
[debug] [2020-12-25T14:44:01.188Z] Platform:      win32
[debug] [2020-12-25T14:44:01.188Z] Node Version:  v10.16.3
[debug] [2020-12-25T14:44:01.190Z] Time:          Fri Dec 25 2020 15:44:01 GMT+0100 (GMT+01:00)
[debug] [2020-12-25T14:44:01.190Z] ----------------------------------------------------------------------
[debug] [2020-12-25T14:44:01.190Z] 
[debug] [2020-12-25T14:44:01.207Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2020-12-25T14:44:01.208Z] > authorizing via signed-in user
[debug] [2020-12-25T14:44:01.208Z] [iam] checking project stocktrackertest-e51fc for permissions ["cloudfunctions.functions.create","cloudfunctions.functions.delete","cloudfunctions.functions.get","cloudfunctions.functions.list","cloudfunctions.functions.update","cloudfunctions.operations.get","firebase.projects.get"]
[debug] [2020-12-25T14:44:01.214Z] >>> HTTP REQUEST POST https://cloudresourcemanager.googleapis.com/v1/projects/stocktrackertest-e51fc:testIamPermissions  
 {"permissions":["cloudfunctions.functions.create","cloudfunctions.functions.delete","cloudfunctions.functions.get","cloudfunctions.functions.list","cloudfunctions.functions.update","cloudfunctions.operations.get","firebase.projects.get"]}
[debug] [2020-12-25T14:44:02.168Z] <<< HTTP RESPONSE 200 {"content-type":"application/json; charset=UTF-8","vary":"X-Origin, Referer, Origin,Accept-Encoding","date":"Fri, 25 Dec 2020 14:44:02 GMT","server":"ESF","cache-control":"private","x-xss-protection":"0","x-frame-options":"SAMEORIGIN","x-content-type-options":"nosniff","server-timing":"gfet4t7; dur=629","alt-svc":"h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\"","accept-ranges":"none","transfer-encoding":"chunked"}
[debug] [2020-12-25T14:44:02.172Z] >>> HTTP REQUEST POST https://iam.googleapis.com/v1/projects/stocktrackertest-e51fc/serviceAccounts/stocktrackertest-e51fc@appspot.gserviceaccount.com:testIamPermissions  
 {"permissions":["iam.serviceAccounts.actAs"]}
[debug] [2020-12-25T14:44:03.157Z] <<< HTTP RESPONSE 200 {"content-type":"application/json; charset=UTF-8","vary":"X-Origin, Referer, Origin,Accept-Encoding","date":"Fri, 25 Dec 2020 14:44:03 GMT","server":"ESF","cache-control":"private","x-xss-protection":"0","x-frame-options":"SAMEORIGIN","x-content-type-options":"nosniff","alt-svc":"h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\"","accept-ranges":"none","transfer-encoding":"chunked"}
[debug] [2020-12-25T14:44:03.423Z] TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undefined
    at validateString (internal/validators.js:125:11)
    at Object.join (path.js:427:7)
    at Object.checkFunctionsSDKVersion [as fn] (C:\Users\Eduard\AppData\Roaming\npm\node_modules\firebase-tools\lib\checkFirebaseSDKVersion.js:34:28)
    at Command.<anonymous> (C:\Users\Eduard\AppData\Roaming\npm\node_modules\firebase-tools\lib\command.js:164:30)
    at Generator.next (<anonymous>)
    at fulfilled (C:\Users\Eduard\AppData\Roaming\npm\node_modules\firebase-tools\lib\command.js:5:58)
    at process._tickCallback (internal/process/next_tick.js:68:7)
[error] 
[error] Error: An unexpected error has occurred.
[debug] [2020-12-25T14:50:06.203Z] ----------------------------------------------------------------------
[debug] [2020-12-25T14:50:06.207Z] Command:       C:\Program Files\nodejs\node.exe C:\Users\Eduard\AppData\Roaming\npm\node_modules\firebase-tools\lib\bin\firebase.js deploy --only functions
[debug] [2020-12-25T14:50:06.207Z] CLI Version:   8.12.1
[debug] [2020-12-25T14:50:06.207Z] Platform:      win32
[debug] [2020-12-25T14:50:06.208Z] Node Version:  v10.16.3
[debug] [2020-12-25T14:50:06.210Z] Time:          Fri Dec 25 2020 15:50:06 GMT+0100 (GMT+01:00)
[debug] [2020-12-25T14:50:06.210Z] ----------------------------------------------------------------------
[debug] [2020-12-25T14:50:06.210Z] 
[debug] [2020-12-25T14:50:06.227Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2020-12-25T14:50:06.228Z] > authorizing via signed-in user
[debug] [2020-12-25T14:50:06.228Z] [iam] checking project stocktrackertest-e51fc for permissions ["cloudfunctions.functions.create","cloudfunctions.functions.delete","cloudfunctions.functions.get","cloudfunctions.functions.list","cloudfunctions.functions.update","cloudfunctions.operations.get","firebase.projects.get"]
[debug] [2020-12-25T14:50:06.234Z] >>> HTTP REQUEST POST https://cloudresourcemanager.googleapis.com/v1/projects/stocktrackertest-e51fc:testIamPermissions  
 {"permissions":["cloudfunctions.functions.create","cloudfunctions.functions.delete","cloudfunctions.functions.get","cloudfunctions.functions.list","cloudfunctions.functions.update","cloudfunctions.operations.get","firebase.projects.get"]}
[debug] [2020-12-25T14:50:07.035Z] <<< HTTP RESPONSE 200 {"content-type":"application/json; charset=UTF-8","vary":"X-Origin, Referer, Origin,Accept-Encoding","date":"Fri, 25 Dec 2020 14:50:07 GMT","server":"ESF","cache-control":"private","x-xss-protection":"0","x-frame-options":"SAMEORIGIN","x-content-type-options":"nosniff","server-timing":"gfet4t7; dur=579","alt-svc":"h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\"","accept-ranges":"none","transfer-encoding":"chunked"}
[debug] [2020-12-25T14:50:07.039Z] >>> HTTP REQUEST POST https://iam.googleapis.com/v1/projects/stocktrackertest-e51fc/serviceAccounts/stocktrackertest-e51fc@appspot.gserviceaccount.com:testIamPermissions  
 {"permissions":["iam.serviceAccounts.actAs"]}
[debug] [2020-12-25T14:50:07.884Z] <<< HTTP RESPONSE 200 {"content-type":"application/json; charset=UTF-8","vary":"X-Origin, Referer, Origin,Accept-Encoding","date":"Fri, 25 Dec 2020 14:50:08 GMT","server":"ESF","cache-control":"private","x-xss-protection":"0","x-frame-options":"SAMEORIGIN","x-content-type-options":"nosniff","alt-svc":"h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\"","accept-ranges":"none","transfer-encoding":"chunked"}
[debug] [2020-12-25T14:50:08.103Z] TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undefined
    at validateString (internal/validators.js:125:11)
    at Object.join (path.js:427:7)
    at Object.checkFunctionsSDKVersion [as fn] (C:\Users\Eduard\AppData\Roaming\npm\node_modules\firebase-tools\lib\checkFirebaseSDKVersion.js:34:28)
    at Command.<anonymous> (C:\Users\Eduard\AppData\Roaming\npm\node_modules\firebase-tools\lib\command.js:164:30)
    at Generator.next (<anonymous>)
    at fulfilled (C:\Users\Eduard\AppData\Roaming\npm\node_modules\firebase-tools\lib\command.js:5:58)
    at process._tickCallback (internal/process/next_tick.js:68:7)
[error] 
[error] Error: An unexpected error has occurred.
[debug] [2020-12-25T14:50:25.536Z] ----------------------------------------------------------------------
[debug] [2020-12-25T14:50:25.539Z] Command:       C:\Program Files\nodejs\node.exe C:\Users\Eduard\AppData\Roaming\npm\node_modules\firebase-tools\lib\bin\firebase.js deploy --only functions
[debug] [2020-12-25T14:50:25.540Z] CLI Version:   8.12.1
[debug] [2020-12-25T14:50:25.540Z] Platform:      win32
[debug] [2020-12-25T14:50:25.540Z] Node Version:  v10.16.3
[debug] [2020-12-25T14:50:25.542Z] Time:          Fri Dec 25 2020 15:50:25 GMT+0100 (GMT+01:00)
[debug] [2020-12-25T14:50:25.542Z] ----------------------------------------------------------------------
[debug] [2020-12-25T14:50:25.543Z] 
[debug] [2020-12-25T14:50:25.559Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2020-12-25T14:50:25.560Z] > authorizing via signed-in user
[debug] [2020-12-25T14:50:25.561Z] [iam] checking project stocktrackertest-e51fc for permissions ["cloudfunctions.functions.create","cloudfunctions.functions.delete","cloudfunctions.functions.get","cloudfunctions.functions.list","cloudfunctions.functions.update","cloudfunctions.operations.get","firebase.projects.get"]
[debug] [2020-12-25T14:50:25.566Z] >>> HTTP REQUEST POST https://cloudresourcemanager.googleapis.com/v1/projects/stocktrackertest-e51fc:testIamPermissions  
 {"permissions":["cloudfunctions.functions.create","cloudfunctions.functions.delete","cloudfunctions.functions.get","cloudfunctions.functions.list","cloudfunctions.functions.update","cloudfunctions.operations.get","firebase.projects.get"]}
[debug] [2020-12-25T14:50:26.382Z] <<< HTTP RESPONSE 200 {"content-type":"application/json; charset=UTF-8","vary":"X-Origin, Referer, Origin,Accept-Encoding","date":"Fri, 25 Dec 2020 14:50:27 GMT","server":"ESF","cache-control":"private","x-xss-protection":"0","x-frame-options":"SAMEORIGIN","x-content-type-options":"nosniff","server-timing":"gfet4t7; dur=576","alt-svc":"h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\"","accept-ranges":"none","transfer-encoding":"chunked"}
[debug] [2020-12-25T14:50:26.386Z] >>> HTTP REQUEST POST https://iam.googleapis.com/v1/projects/stocktrackertest-e51fc/serviceAccounts/stocktrackertest-e51fc@appspot.gserviceaccount.com:testIamPermissions  
 {"permissions":["iam.serviceAccounts.actAs"]}
[debug] [2020-12-25T14:50:27.213Z] <<< HTTP RESPONSE 200 {"content-type":"application/json; charset=UTF-8","vary":"X-Origin, Referer, Origin,Accept-Encoding","date":"Fri, 25 Dec 2020 14:50:27 GMT","server":"ESF","cache-control":"private","x-xss-protection":"0","x-frame-options":"SAMEORIGIN","x-content-type-options":"nosniff","alt-svc":"h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\"","accept-ranges":"none","transfer-encoding":"chunked"}
[debug] [2020-12-25T14:50:27.451Z] TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undefined
    at validateString (internal/validators.js:125:11)
    at Object.join (path.js:427:7)
    at Object.checkFunctionsSDKVersion [as fn] (C:\Users\Eduard\AppData\Roaming\npm\node_modules\firebase-tools\lib\checkFirebaseSDKVersion.js:34:28)
    at Command.<anonymous> (C:\Users\Eduard\AppData\Roaming\npm\node_modules\firebase-tools\lib\command.js:164:30)
    at Generator.next (<anonymous>)
    at fulfilled (C:\Users\Eduard\AppData\Roaming\npm\node_modules\firebase-tools\lib\command.js:5:58)
    at process._tickCallback (internal/process/next_tick.js:68:7)
[error] 
[error] Error: An unexpected error has occurred.
